lisa/schema.py

```diff
@@ -659,35 +672,31 @@ class Environment:
 
     def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         self.nodes: Optional[List[Union[LocalNode, RemoteNode]]] = None
-        if self.nodes_raw is not None:
-            self.nodes = []
-            for node_raw in self.nodes_raw:
-                node_type = node_raw[constants.TYPE]
-                if node_type == constants.ENVIRONMENTS_NODES_LOCAL:
-                    node: Union[
-                        LocalNode, RemoteNode
-                    ] = LocalNode.schema().load(  # type:ignore
-                        node_raw
-                    )
-                    if self.nodes is None:
-                        self.nodes = []
-                    self.nodes.append(node)
-                elif node_type == constants.ENVIRONMENTS_NODES_REMOTE:
-                    node = RemoteNode.schema().load(node_raw)  # type:ignore
-                    if self.nodes is None:
-                        self.nodes = []
-                    self.nodes.append(node)
-                elif node_type == constants.ENVIRONMENTS_NODES_REQUIREMENT:
-                    original_req: NodeSpace = NodeSpace.schema().load(  # type:ignore
-                        node_raw
-                    )
-                    expanded_req = original_req.expand_by_node_count()
-                    if self.nodes_requirement is None:
-                        self.nodes_requirement = []
-                    self.nodes_requirement.extend(expanded_req)
-                else:
-                    raise LisaException(f"unknown node type '{node_type}': {node_raw}")
-            self.nodes_raw = None
+        if not self.nodes_raw:
+            return
+        nodes_final: List[Union[LocalNode, RemoteNode]] = []
+        for node in self.nodes_raw:
+            node_type = node[constants.TYPE]
```

Comment:
I hit an exception here when run test on platform.     
node_type = node[constants.TYPE]
TypeError: string indices must be integers